### PR TITLE
Bump OpenTelemetry Collector dependencies to v0.71.0

### PR DESCRIPTION
**Pull request by gbbr**

### CODE REVIEW

File: pkg/otlp/model/go.mod [hunk 1]
```diff
@@ -12,7 +12,7 @@ require (
 	github.com/patrickmn/go-cache v2.1.0+incompatible
 	github.com/stretchr/testify v1.8.1
 	go.opentelemetry.io/collector/pdata v1.0.0-rc4
```
Comment (mx-psi) [inline pkg/otlp/model/go.mod:14]: needs an update here too
```suggestion
	go.opentelemetry.io/collector/pdata v1.0.0-rc5
```

File: pkg/trace/go.mod [hunk 2]
```diff
@@ -30,7 +30,7 @@ require (
 	github.com/tinylib/msgp v1.1.6
 	github.com/vmihailenco/msgpack/v4 v4.3.12
 	go.opentelemetry.io/collector/pdata v1.0.0-rc4
```
Comment (mx-psi) [inline pkg/trace/go.mod:32]:
```suggestion
	go.opentelemetry.io/collector/pdata v1.0.0-rc5
```

**Review by ahmed-mez — APPROVED** [review]

File: releasenotes/notes/bump-otel-71-4e4090a67d42a047.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    The OTLP ingest endpoint now supports the same settings and protocol as the OpenTelemetry Collector OTLP receiver v0.71.0
```
Comment (cswatt) [inline releasenotes/notes/bump-otel-71-4e4090a67d42a047.yaml:11]:
```suggestion
    The OTLP ingest endpoint now supports the same settings and protocol as the OpenTelemetry Collector OTLP receiver v0.71.0.
```

**Review by cswatt — APPROVED** [review]
added one tiny suggestion to release note, but approving as to not block

File: pkg/trace/go.mod (hunk 2 above)
Comment (mx-psi) [reply pkg/trace/go.mod:32]: This one still needs an update

Comment (gbbr) [reply pkg/trace/go.mod]: Right, my bad. Should be good now.

**Comment by mx-psi** [comment]
Looks like this needs `invoke tidy-all` and `invoke generate-licenses`, otherwise LGTM

**Comment by gbbr** [comment]
I actually ran both of those and got no output. Let me try again. Sorry for the back and forth.

**Review by mx-psi — APPROVED** [review]
